### ISSUE TITLE
Populate page metadata, social media metadata

### ISSUE THREAD

**Issue by mattxwang** [uclaacm/website#116]
One of the follies of Single Page Applications (SPAs) is that you need to do a bit of extra work to make the website play nicely with web crawlers (ex google's preview) and social media embeds (ex links from Facebook, Twitter, Discord).

Luckily, this is a relatively solved problem. Given the (so far) small-ish number of pages on our site, this is something that we can do mostly manually, with some minor automation if needed.

There are several disjoint things we need to consider:

1. Properly filling out `<meta>` tags in the document head.  One defacto standard is [`react-helmet`](https://github.com/nfl/react-helmet).
2. Properly dealing with [Open Graph](https://ogp.me/) metadata, commonly used for social media embeds (and developed by Facebook). This is quite hard, since these crawlers (unlike Google) tend to not run JS bundles.
3. Customizing for certain networks, such as [Twitter Cards](https://developer.twitter.com/en/docs/twitter-for-websites/cards/overview/abouts-cards) - this is a nice-to-have, and not a requirement for this ticket

## Social Media Optimization with JavaScript

Open Graph is by far the trickiest part of this situation, since SPAs by nature don't work properly without JavaScript, and most crawlers don't run JS when they crawl a site. To remedy this, there are a few options; here they are in orders of complexity:

1. adding a default OG metadata to the public `index.html`. Not great, but better than what we have right now.
2. investigate a hacky solution like [`react-snap`](https://github.com/stereobooster/react-snap), which prerenders + crawls our webapp to create HTML pages for crawlers.
3. use one of the frameworks designed to solve this problem / take advantage of Server Side Rendering (SSR): [Next.js](https://nextjs.org/) and [Gatsby](https://www.gatsbyjs.com/) are likely the two best contenders.

I personally would *absolutely love* to do the last option, though it has its own ramifications around maintainability (we need to teach people Gatsby/Next) and the likely insane amount of merge conflicts this would generate with other PRs.

Also would encourage you to ask around / do your own research around this problem - I (Matt) am definitely not as in the loop as other web developers on the perfect way to resolve this problem. Some possible ACM-adjacent people to discuss this with:

* ACM Hack is a big fan of Gatsby, they use it for [their website](https://github.com/uclaacm/hack.uclaacm.com) and [HOTH](https://github.com/uclaacm/hoth.uclaacm.com)
* Creative Labs uses Next frequently, including for [their website](https://github.com/UCLA-Creative-Labs/sunshine)

This is one of those tickets that seems very simple, but is actually quite complicated/interesting!

**Comment by mattxwang** [comment]
Blocked by SSG rework!
